rotkehlchen/tests/unit/decoders/test_uniswapv2.py

```diff
@@ -0,0 +1,324 @@
+                ],
+            ), EthereumTxReceiptLog(
+                log_index=309,
+                data=hexstring_to_bytes('0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000015f8d0c01dad329c0000000000000000000000000000000000000000204fce5e3e250261100000000000000000000000000000000000000000000000000000000000000000000000'),  # noqa: E501
+                address=string_to_evm_address('0xB4e16d0168e52d35CaCD2c6185b44281Ec28C9Dc'),
+                removed=False,
+                topics=[
+                    hexstring_to_bytes('0xd78ad95fa46c994b6551d0da85fc275fe613ce37657fb8d5e3d130840159d822'),  # noqa: E501
+                    hexstring_to_bytes('0x0000000000000000000000007a250d5630b4cf539739df2c5dacb4c659f2488d'),  # noqa: E501
+                    hexstring_to_bytes('0x0000000000000000000000003cadf2ca458376a6a5fea2ef3612346037d5a787'),  # noqa: E501
+                ],
+            ),
+        ],
+    )
+
+    internal_tx = EthereumInternalTransaction(
+        parent_tx_hash=evmhash,
+        trace_id=27,
+        timestamp=Timestamp(1646375440),
+        block_number=14351442,
+        from_address='0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D',
+        to_address='0x3CAdf2cA458376a6a5feA2EF3612346037D5A787',
+        value=FVal('0.008104374914845978') * EXP18,
+    )
+
+    dbethtx = DBEthTx(database)
+    decoder = EVMTransactionDecoder(
+        database=database,
+        ethereum_manager=ethereum_manager,
+        eth_transactions=eth_transactions,
+        msg_aggregator=msg_aggregator,
+    )
+    with database.user_write() as cursor:
+        dbethtx.add_ethereum_transactions(cursor, [transaction], relevant_address=None)
+        dbethtx.add_ethereum_internal_transactions(cursor, [internal_tx], relevant_address='0x3CAdf2cA458376a6a5feA2EF3612346037D5A787')  # noqa: E501
+        events = decoder.decode_transaction(cursor, transaction=transaction, tx_receipt=receipt)
+
+    assert len(events) == 4
+    expected_events = [
+        HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(tx_hex),
+            sequence_index=0,
+            timestamp=1646375440000,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(
+                amount=FVal(0.00393701451),
+                usd_value=ZERO,
+            ),
+            location_label='0x3CAdf2cA458376a6a5feA2EF3612346037D5A787',
+            notes='Burned 0.00393701451 ETH in gas from 0x3CAdf2cA458376a6a5feA2EF3612346037D5A787',  # noqa: E501
+            counterparty=CPT_GAS,
+        ), HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(tx_hex),
+            sequence_index=1,
+            timestamp=1646375440000,
+            location=Location.BLOCKCHAIN,
```

Comment:
Hmm ... but how is this going to be treated at the accountant? This receiving is not gonna be seen as part of the trade

Reply:
I was thinking about that. This is eth that you send as you are converting all and then receive back part of it. Technically you are not swapping it or doing nothing with it and after the transaction is as if you just swapped x-received amount after sending x